src/permissions/plugins/heroku/index.ts

```diff
@@ -0,0 +1,112 @@
+import HerokuClient from 'heroku-client';
+
+import { MessageBuilder } from '../../../MessageBuilder.js';
+import { RepositoryConfig, TeamConfig } from '../../types.js';
+import { Plugin } from '../Plugin.js';
+import { HEROKU_MAGIC_ADMIN, HEROKU_TOKEN, SHERIFF_GSUITE_DOMAIN } from '../../../constants.js';
+import { IS_DRY_RUN } from '../../../helpers.js';
+import chalk from 'chalk';
+
+interface HerokuCollab {
+  id: string;
+  user: {
+    email: string;
+  };
+}
+
+interface HerokuMember {
+  role: 'collaborator' | 'admin';
+  user: {
+    email: string;
+  };
+}
+
+interface HerokuApp {
+  locked: boolean;
+}
+
+class HerokuPlugin implements Plugin {
+  private client = new HerokuClient({
+    token: HEROKU_TOKEN,
+  });
+
+  private emailSame = (e1: string, e2: string) => e1.toLowerCase() === e2.toLowerCase();
+
+  handleRepo = async (repo: RepositoryConfig, teams: TeamConfig[], builder: MessageBuilder) => {
+    const { heroku } = repo;
+    if (!heroku) return;
+
+    const collaborators = (
+      (await this.client.get(`/teams/apps/${heroku.app_name}/collaborators`)) as HerokuCollab[]
+    ).filter(
+      (c) => !c.user.email.endsWith('@herokumanager.com') && c.user.email !== HEROKU_MAGIC_ADMIN,
+    );
+
+    const teamAdmins = (
+      (await this.client.get(`/teams/${heroku.team_name}/members`)) as HerokuMember[]
+    ).filter((m) => m.role === 'admin');
+
+    const app = (await this.client.get(`/teams/apps/${heroku.app_name}`)) as HerokuApp;
+    // Just lock every app we control
+    if (!app.locked && !IS_DRY_RUN) {
+      await this.client.patch(`/teams/apps/${heroku.app_name}`, {
+        body: {
+          locked: true,
+        },
+      });
+    }
```

Comment:
note for posterity: "locking" apps means that the app can't be joined by other team members: https://devcenter.heroku.com/articles/platform-api-reference#team-app.

this is something we want for all Heroku apps because perms should exclusively be controlled via Sheriff. The first time that Sheriff runs on a given app, it will lock it down if it isn't already.